test/hotspot/jtreg/gc/metaspace/TestCapacityUntilGCWrapAround.java

```diff
@@ -41,19 +40,30 @@
 
 public class TestCapacityUntilGCWrapAround {
     private static long MB = 1024 * 1024;
-    private static long GB = 1024 * MB;
-    private static long MAX_UINT = 4 * GB - 1; // On 32-bit platforms
-
-    public static void main(String[] args) {
-        WhiteBox wb = WhiteBox.getWhiteBox();
 
+    private static void incMetaspaceCapacityUntilGCTest(WhiteBox wb) {
         long before = wb.metaspaceCapacityUntilGC();
-        // Now force possible overflow of capacity_until_GC.
-        long after = wb.incMetaspaceCapacityUntilGC(MAX_UINT);
+        long after = wb.incMetaspaceCapacityUntilGC(100 * MB);
```

Comment:
The test makes only sense with 4G increment. AFAIU the counter is supposed to overflow, and cap at MaxMetaspaceSize instead.